core/libp2p/muxer/yamux/yamuxed_connection.hpp

```diff
@@ -0,0 +1,238 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_YAMUXED_CONNECTION_IMPL_HPP
+#define KAGOME_YAMUXED_CONNECTION_IMPL_HPP
+
+#include <functional>
+#include <map>
+#include <queue>
+
+#include <boost/asio/streambuf.hpp>
+#include <boost/system/error_code.hpp>
+#include "common/buffer.hpp"
+#include "common/logger.hpp"
+#include "libp2p/muxer/muxer_adaptor.hpp"
+#include "libp2p/muxer/yamux/yamux_config.hpp"
+#include "libp2p/muxer/yamux/yamux_stream_parameters.hpp"
+#include "libp2p/stream/stream.hpp"
+#include "libp2p/transport/connection.hpp"
+#include "libp2p/transport/muxed_connection.hpp"
+#include "libp2p/connection/stream.hpp"
+
+namespace libp2p::stream {
+  class YamuxStream;
+}
+
+namespace libp2p::muxer {
+  struct YamuxFrame;
```

Comment:
why is it struct? is it supposed to have only state?

Reply:
yes. I did not touch this code, it is yamux implementation